tests/nix-shell.sh

```diff
@@ -96,6 +96,8 @@ nix develop -f "$shellDotNix" shellDrv -c echo foo |& grep -q foo
 # Test 'nix print-dev-env'.
 [[ $(nix print-dev-env -f "$shellDotNix" shellDrv --json | jq -r .variables.arr1.value[2]) = '3 4' ]]
 
+set +u # TODO fix?
+
 source <(nix print-dev-env -f "$shellDotNix" shellDrv)
```

Comment:
`print-dev-env` outputs bash which is not compliant with these flags. I didn't want to change the behavior of Nix as part of a test cleanup however.

Reply:
Opened https://github.com/NixOS/nix/issues/7951 to track that